Sources/PackageModel/UserToolchain.swift

```diff
@@ -117,15 +117,55 @@ public final class UserToolchain: Toolchain {
         return toolPath
       }
     }
+
+    if useXcrun {
```

Comment:
I would like to introduce this approach as a fall-back from the beginning, and then makes it the default (like Clang does) if everything is working as expected.